### DISCUSSION TITLE
ESP32-S3 random resets with rst:0x8 (TG1WDT_SYS_RST),boot:0x8 (SPI_FAST_FLASH_BOOT)

### DISCUSSION THREAD

**Discussion by gampam2000** [micropython/micropython#15247] [ESP32]
Hi,

on my ESP32-S3-WROOM-1-N8R2 i get random resets with this output:

```
ESP-ROM:esp32s3-20210327
Build:Mar 27 2021
rst:0x8 (TG1WDT_SYS_RST),boot:0x8 (SPI_FAST_FLASH_BOOT)
Saved PC:0x42010683
SPIWP:0xee
mode:DIO, clock div:1
load:0x3fce3810,len:0xf8c
load:0x403c9700,len:0xb3c
load:0x403cc700,len:0x2dd4
entry 0x403c989c
W (29) boot.esp32s3: PRO CPU has been reset by WDT.
W (29) boot.esp32s3: APP CPU has been reset by WDT.
```

MPY version : v1.23.0 on 2024-06-08
IDF version : v5.2
Machine     : Generic ESP32S3 module with ESP32S3

It seems to happen when I access/process a rather large array with 536 entry's which hold dictionary objects. So i have 536 dicts in an array. I can not pinpoint the exact location because it happens rather random. 

How can I debug this? 
Is it possible to extend the ESP-IDF watchdog from micropython?

**Comment by summerfind** [comment]
this one solved my problem. 
https://github.com/espressif/arduino-esp32/issues/7072

**Reply by gampam2000** [reply]
Thanks for the link.
You mean you used a pin for SPI that was used for the SPIRAM? And changing it solved the problem?
What GPIO created the issue?

I use SPI1 for a Display and SPI2 for Lora. According to the ESP32-S3 Datasheet I can not use GPIOs:
35,36,37. I do not use them. What module/board do you use?

My Display uses:
```
LED_RESET = Pin(5, Pin.OUT)
OLED_DC = Pin(6, Pin.OUT)
OLED_CS = Pin(7, Pin.OUT)
OLED_SPI_SCK = Pin(15)
OLED_SPI_MOSI = Pin(16)
OLED_BACKLIGHT = Pin(2, Pin.OUT)    
```

Lora uses: 
```
LORA_SPI_MOSI = 41
LORA_SPI_SCK = 40
LORA_SPI_MISO = 38
LORA_NSS = 9                    # CHIP Select
LORA_BUSY = 13                  # in Driver it is called GPIO
LORA_DIO1 = 45                  # IRQ
LORA_RESET = 17 
```

Thanks for our help

**Comment by gampam2000** [comment]
I still have the same issues that at random instances the ESP32S3 resets, even with the latest micropython:

```
ESP-ROM:esp32s3-20210327
Build:Mar 27 2021
rst:0x8 (TG1WDT_SYS_RST),boot:0x8 (SPI_FAST_FLASH_BOOT)
Saved PC:0x42010bff
SPIWP:0xee
mode:DIO, clock div:1
load:0x3fce3820,len:0x105c
load:0x403c9700,len:0x4
load:0x403c9704,len:0xbd8
load:0x403cc700,len:0x2e34
entry 0x403c989c
W (31) boot.esp32s3: PRO CPU has been reset by WDT.
W (31) boot.esp32s3: APP CPU has been reset by WDT.
Micropython v1.24.0-preview.278.g5e692d046.dirty
```
How can I debug this?

**Reply by liudr** [reply]
You mentioned extending WDT timer. Have you tried that? I don't know if the following helps:

```
from machine import WDT
wdt=WDT(timeout=milli_seconds)	# WDT timeout uses milliseconds

```
Give it something like a minute. I set it to 300 seconds so if my board stops feeding it due to code stuck or non-recovered critical failure, it reboots itself after 300 seconds.
Then you must periodically call wdt.feed().
I thought micropython doesn't use WDT. Also, you're running a preview version, any reasons you can't run a release version like 1.23?

**Comment by gampam2000** [comment]
Hi, thanks for this suggestion. I allready tried to use the micropython watchdog, but it seems to be completely independent from the FreeRTOS Watchdog that is triggered. 

I also tried the release versions but I still get the random resets with watchdog timeouts. Sometimes it runs for hours before i get a reset. My question is how i can debug this?

**Reply by liudr** [reply]
Maybe post more complete printouts on esp32.com forum since the messages are from the underlying ESP-IDF, not really MicroPython. RTOS is below ESP-IDF. It's like this:
RTOS->ESP-IDF->MicroPython->your code.
